tapir/coop/forms.py

```diff
@@ -247,3 +249,67 @@ class Meta:
+                  "cancellation_reason",
+                  "cancellation_date", 
+                  "coop_buys_shares_back",
+                  "willing_to_gift_shares_to_coop",
+                  "transfering_shares_to",
+                  "paid_out",
+        ]
+        widgets = {"cancellation_date": DateInputTapir()}
+
+    def clean(self):
+        cleaned_data = super().clean()
+        share_owner = cleaned_data.get("share_owner")
+        coop_buys_shares_back = cleaned_data.get("coop_buys_shares_back")
+        willing_to_gift_shares_to_coop = cleaned_data.get("willing_to_gift_shares_to_coop")
+        transfering_shares_to = cleaned_data.get("transfering_shares_to")
+        paid_out = cleaned_data.get("paid_out")
+        cancellation_date = cleaned_data.get("cancellation_date")
+        errmsg = _("Please take only one choice.")
+
+        if self.instance.pk is None:
+            for alreadyResignedMember in self.already_resigned.values("share_owner"):
+                if alreadyResignedMember['share_owner'] == share_owner.id:
+                    self.add_error("share_owner", ValidationError(
+                    _("This member is already resigned.")
+                    ))
+                    break
+
+        if coop_buys_shares_back and willing_to_gift_shares_to_coop:
+            self.add_error("coop_buys_shares_back", errmsg)
+            self.add_error("willing_to_gift_shares_to_coop", errmsg)
+        elif transfering_shares_to != None and (coop_buys_shares_back or willing_to_gift_shares_to_coop):
+            self.add_error("transfering_shares_to", errmsg)
+            if coop_buys_shares_back:
+                self.add_error("coop_buys_shares_back", errmsg)
+            elif willing_to_gift_shares_to_coop:
+                self.add_error("willing_to_gift_shares_to_coop", errmsg)
+        if transfering_shares_to is not None:
+            if transfering_shares_to.share_owner == share_owner:
+                self.add_error("transfering_shares_to", ValidationError(
+                    _("Sender and receiver of tranfering the share(s) cannot be the same.")
+                    ))  
+        if (transfering_shares_to != None and paid_out) or (willing_to_gift_shares_to_coop and paid_out) :
+            self.add_error("paid_out", ValidationError(_("Cannot pay out, because shares have been gifted.")
+                    ))
+        if transfering_shares_to == None and not willing_to_gift_shares_to_coop and not coop_buys_shares_back:
+            self.add_error("transfering_shares_to", ValidationError(_("Please make a least one choice.")))
+            self.add_error("willing_to_gift_shares_to_coop", ValidationError(_("Please make a least one choice.")))
+            self.add_error("coop_buys_shares_back", ValidationError(_("Please make a least one choice.")))
+        if coop_buys_shares_back:
+            self.instance.pay_out_day = cancellation_date + relativedelta(day=31, month=12, years=3)
```

Comment:
Is this addition of 31 days correct? What you are trying to achieve? Maybe there is a better way to do it.

Reply:
What do you mean by "a better way"? If the coop buys back the share, the date of the pay out day is 31/12 + 3 years. So this should correct, or?

Reply:
I see, so it might be for example in the middle of a month. Then it's fine like that.